Include `chrono` for `high_resolution_clock`

<!-- Thank you for contributing to osquery! -->

# Bug report

I work on Microsoft Visual C++ testing, where we regularly build popular open-source projects, including yours, with development builds of our compiler and libraries to detect and prevent shipping regressions that would affect you. This also allows us to provide advance notice of breaking changes, which is the case here.

Recently https://github.com/microsoft/STL/pull/5105, revealed an issue in OSQuery.

Compiler error with this STL change:
```c++
C:\gitP\facebook\osquery\osquery\config\packs.cpp(65,20): error C2039: 'high_resolution_clock': is not a member of 'std::chrono' 
```

Affected code:
https://github.com/osquery/osquery/blob/aa60bbdf2ea8a1ab160f7f1b40db658a8a3770d3/osquery/config/packs.cpp#L65

This was assuming that including makes the chrono::high_resolution_clock type available, which is not guaranteed by the Standard. You need to explicitly include the chrono header in this file, and the error will be resolved.

### What operating system and version are you using?
Windows 11

<!--
You can run the command below. Paste result within a code block.

osqueryi --line "SELECT version, build, platform FROM os_version;"
-->

### What version of osquery are you using?
latest commit

Build log: 
[osquery.txt.log](https://github.com/user-attachments/files/17913936/osquery.txt.log)

--- a/osquery/config/packs.cpp
+++ b/osquery/config/packs.cpp
@@ -9,6 +9,7 @@
 
 #include <algorithm>
 #include <mutex>
+#include <chrono>
 #include <random>
 
 #include <osquery/config/packs.h>

